File: src/main/java/jp/aoichaan0513/A_TosoGame_Live/API/Manager/GameManager.java
package jp.aoichaan0513.A_TosoGame_Live.API.Manager;

import jp.aoichaan0513.A_TosoGame_Live.Main;
import jp.aoichaan0513.A_TosoGame_Live.Runnable.GameRunnable;
import org.bukkit.scheduler.BukkitTask;

public class GameManager {

    private static GameState game = GameState.NONE;

    private static BukkitTask timer = null;

    public static void startGame(int countDown, int gameTime) {
        if (timer != null) return;
        timer = new GameRunnable(countDown, gameTime).runTaskTimer(Main.getInstance(), 0L, 20L);
        GameManager.setGameState(GameManager.GameState.READY);
    }

    public static void endGame() {
        if (timer == null) return;
        timer.cancel();
        timer = null;
        GameManager.setGameState(GameManager.GameState.END);
        MissionManager.endMission();
    }

    public static boolean isGame(GameState game) {
        return GameManager.game == game;
    }

    public static boolean isGame() {
        return game == GameState.READY || game == GameState.GAME;
    }

    public static void setGameState(GameState game) {
        GameManager.game = game;
    }

    public static GameState getGameState() {
        return game;
    }

    public enum GameState {
        READY,
        GAME,
        END,
        NONE
    }
}

File: src/main/kotlin/jp/aoichaan0513/A_TosoGame_Live/API/Manager/GameManager.kt
package jp.aoichaan0513.A_TosoGame_Live.API.Manager

import jp.aoichaan0513.A_TosoGame_Live.Main
import jp.aoichaan0513.A_TosoGame_Live.Mission.MissionManager
import jp.aoichaan0513.A_TosoGame_Live.OPGame.Dice
import jp.aoichaan0513.A_TosoGame_Live.OPGame.OPGameManager
import jp.aoichaan0513.A_TosoGame_Live.Runnable.GameRunnable
import org.bukkit.Bukkit
import org.bukkit.scheduler.BukkitTask

class GameManager {
    companion object {

        var gameState = GameState.NONE
        private var timer: BukkitTask? = null

        fun startGame(countDown: Int, gameTime: Int) {
            if (timer != null) return
            timer = GameRunnable(countDown, gameTime).runTaskTimer(Main.pluginInstance, 0L, 20L)
            gameState = GameState.READY
        }

        fun endGame() {
            if (timer == null) return
            timer!!.cancel()
            timer = null
            gameState = GameState.END

            MissionManager.endMissions()
            Dice.end()
            OPGameManager.endOPGame(Bukkit.getConsoleSender())
        }

        fun isGame(): Boolean {
            return gameState == GameState.READY || gameState == GameState.GAME
        }

        fun isGame(game: GameState): Boolean {
            return gameState == game
        }
    }

    enum class GameState {
        READY,
        GAME,
        END,
        NONE
    }
}